api/v1alpha1/vspheredistributednetwork_types.go

```diff
@@ -123,12 +123,66 @@ type VlanSpec struct {
 	PrivateVlanID *int32 `json:"privateVlanID,omitempty"`
 }
 
+// MacLimitPolicyType represents the policy type to be used when the MAC address limit is exceeded.
+type MacLimitPolicyType string
+
+const (
+	// MacLimitPolicyAllow indicates that new MAC addresses should still be allowed when the limit is exceeded.
+	MacLimitPolicyAllow MacLimitPolicyType = "allow"
+	// MacLimitPolicyDrop indicates that new MAC addresses should be dropped when the limit is exceeded.
+	MacLimitPolicyDrop MacLimitPolicyType = "drop"
+)
+
+// MacLearningPolicy represents the MAC learning policy configuration.
+type MacLearningPolicy struct {
+	// Enabled indicates whether MAC learning is enabled.
+	Enabled bool `json:"enabled,omitempty"`
+
+	// AllowUnicastFlooding indicates whether to allow flooding of unlearned MAC for ingress traffic.
+	// +optional
+	AllowUnicastFlooding *bool `json:"allowUnicastFlooding,omitempty"`
+
+	// Limit represents the maximum number of MAC addresses that can be learned.
+	// +optional
+	Limit *int32 `json:"limit,omitempty"`
+
+	// LimitPolicy represents the policy to be used when the limit is exceeded.
+	// +optional
+	LimitPolicy *MacLimitPolicyType `json:"limitPolicy,omitempty"`
+}
+
+// MacManagementPolicy represents the MAC management policy configuration.
+type MacManagementPolicy struct {
+	// AllowPromiscuous indicates whether promiscuous mode is enabled. Determines whether or not all
+	// traffic is seen on the port.
+	// +optional
+	AllowPromiscuous *bool `json:"allowPromiscuous,omitempty"`
+
+	// MacChanges indicates whether MAC address changes can be changed.
+	// +optional
+	MacChanges *bool `json:"macChanges,omitempty"`
+
+	// ForgedTransmits indicates whether or not the virtual network adapter should be allowed to send
+	// network traffic with a different MAC address than the one assigned to it.
+	// +optional
+	ForgedTransmits *bool `json:"forgedTransmits,omitempty"`
+
+	// MacLearningPolicy represents the MAC learning policy configuration.
+	// +optional
+	MacLearningPolicy *MacLearningPolicy `json:"macLearningPolicy,omitempty"`
+}
+
 // VSphereDistributedPortConfig represents the port-level configuration for a vSphere Distributed Network
 type VSphereDistributedPortConfig struct {
 	// Vlan represents the VLAN configuration for this port.
 	// If unset, indicates that no VLAN configuration has been retrieved yet for this port.
 	// +optional
 	Vlan *VlanSpec `json:"vlan,omitempty"`
+
+	// MacManagementPolicy represents the MAC management policy configuration for this port.
+	// If unset, indicates that no MAC management policy has been retrieved yet for this port.
+	// +optional
+	MacManagementPolicy *MacManagementPolicy `json:"macManagementPolicy,omitempty"`
```

Comment:
I am not sure about the name, it's not limited to mac related.
Probably better to name it `SecurityManagementPolicy` since it's some configs under the `Security` panel in VC settings.

Reply:
This is directly modeling the VMODL1 APIs.

See https://developer.broadcom.com/xapis/vsphere-web-services-api/latest/vim.dvs.VmwareDistributedVirtualSwitch.MacManagementPolicy.html.